src/runtime/schema.ts

```diff
@@ -175,11 +176,28 @@ export class Schema {
     };
   }
 
+  static _expressionString(expr) : string {
```

Comment:
It looks like this duplicates some work from src/runtime/particle-spec.ts
Lets see if we can put it all in the one place (it'll get more complicated if we start eliminating parens etc).